jdaviz/core/marks.py

```diff
@@ -104,8 +104,17 @@ def set_y_unit(self, unit=None):
                 return
             unit = self.viewer.state.y_display_unit
         unit = u.Unit(unit)
+
         if self.yunit is not None:
-            self.y = (self.y * self.yunit).to_value(unit)
+            if ("spec" in self.viewer.reference or "specviz" in self.viewer.reference_id):
```

Comment:
I don't love the way I coded this check, open to better suggestions.

Reply:
what about `str(self.xunit.physical_type) in ['frequency', 'length']` (ideally it would be great to check the physical type of the yunit as well, but that often seems to come back as "unknown").  

Or maybe cleaner, just have a property for `x_equivalencies` on the mark object that does the following two lines and returns `eqv`.  We can then modify in the future if needed with more logic (if there are ever viewers in jdaviz that don't want to use the spectral density equivalency), and downstream apps can just override the property entirely?

Reply:
Or how about `isinstance(self.viewer, SpectrumViewer)`?  I'm worried that this current implementation is going to cause issues if/when we ever allow renaming viewers.

Reply:
I think the `isinstance` check is probably the right answer.

Reply:
Hmm, actually I think that introduces a circular import. Needs slightly more thinking...